src/BlockTemplatesController.php

```diff
@@ -278,16 +294,25 @@ function( $saved_woo_template ) {
 	 *
 	 * @param string[] $slugs An array of slugs to filter templates by. Templates whose slug does not match will not be returned.
 	 * @param array    $already_found_templates Templates that have already been found, these are customised templates that are loaded from the database.
+	 * @param array    $template_type wp_template or wp_template_part.
 	 *
 	 * @return array Templates from the WooCommerce blocks plugin directory.
 	 */
-	public function get_block_templates_from_woocommerce( $slugs, $already_found_templates ) {
-		$template_files = BlockTemplateUtils::gutenberg_get_template_paths( $this->templates_directory );
+	public function get_block_templates_from_woocommerce( $slugs, $already_found_templates, $template_type = 'wp_template' ) {
+		$directory      = $this->get_templates_directory( $template_type );
+		$template_files = BlockTemplateUtils::gutenberg_get_template_paths( $directory );
 		$templates      = array();
 		foreach ( $template_files as $template_file ) {
+
+			if ( 'wp_template_part' === $template_type ) {
+				$dir_name = self::TEMPLATE_PARTS_DIR_NAME;
+			} else {
+				$dir_name = self::TEMPLATES_DIR_NAME;
+			}
```

Comment:
We should move this check outside of the `foreach` loop.

Reply:
Fixed in 51930edda5955e95497df145add49d629d89c26d.